Src/VimCore/VimBuffer.fs

```diff
@@ -702,7 +702,23 @@ type internal VimBuffer
  
             if _isClosed && not x.IsProcessingInput then
                 _postClosedEvent.Trigger x   
-    
+
+    /// Process from C# script
+    member x.ProcessFromScript (keyInput: KeyInput) =

```

Comment:
Why is this still needed?

Reply:
Why does `Process` not work here? I've thought about this for a bit and can't see what's wrong here. Is it the recursive event handling that you're worried aboutal

Reply:
I may be wrong
Here I think that it is as follows.

1. Type j.
2. An OnKeyInputStart event occurs.
3. Since "e.Handled = true", Vsvim does nothing.
4. "vimBuffer.Process ('k')" is executed in the OnKeyInputStart event.
5. An OnKeyInputStart event occurs.
6. Since "e.Handled = true" is set, for the key of "k" Vsvim does nothing.

Reply:
If I make the following changes, the code will be executed.

```
#r "Microsoft.VisualStudio.CoreUtility.dll"

using Vim;
using Vim.Extensions;

var activeVimBuffer = Vim.ActiveBuffer;
if (activeVimBuffer.IsNone())
{
    Vim.ActiveStatusUtil.OnError(":Can not get VimBuffer");
    return;
}
IVimBuffer vimBuffer = activeVimBuffer.Value;
vimBuffer.KeyInputStart += OnKeyInputStart;

public void OnKeyInputStart(object sender, KeyInputStartEventArgs e)
{
    if (e.KeyInput.Char == 'k')
    {
        return;
    }

    e.Handled = true;
    if (e.KeyInput.Char == 'j')
    {
        var ki = KeyInputUtil.CharToKeyInput('k');
        //vimBuffer.ProcessFromScript(ki);
        vimBuffer.Process(ki); //Work.
    }
    else if (e.KeyInput.Key == VimKey.Escape)
    {
        vimBuffer.KeyInputStart -= OnKeyInputStart;
    }
}

```

Reply:
I think that there is no problem unless `Process` is used in `OnKeyInputStart` event.
However, I think that some users would like to use `Process` in the `OnKeyInputStart` event.

Reply:
I thought for a moment.
And if I modify the code as follows,
I noticed that it works.
So, I remove ProcessFromScript.

```
public void OnKeyInputStart(object sender, KeyInputStartEventArgs e)
{

    e.Handled = true;
    if (e.KeyInput.Char == 'j')
    {
        vimBuffer.KeyInputStart -= OnKeyInputStart;

        var ki = KeyInputUtil.CharToKeyInput('k');
        vimBuffer.Process(ki); //Work.

        vimBuffer.KeyInputStart += OnKeyInputStart;
    }
    else if (e.KeyInput.Key == VimKey.Escape)
    {
        vimBuffer.KeyInputStart -= OnKeyInputStart;
    }
}
```